crates/sui-indexer-alt-framework/src/pipeline/sequential/committer.rs

```diff
@@ -92,8 +90,10 @@ where
         let mut pending: BTreeMap<u64, IndexedCheckpoint<H>> = BTreeMap::new();
         let mut pending_rows = 0;
 
-        info!(pipeline = H::NAME, ?watermark, "Starting committer");
+        info!(pipeline = H::NAME, "Starting committer");
 
+        // Cache current watermark in case of db txn failures.
```

Comment:
I initially missed the significance of the fact that we might fail to write a whole batch (and its watermark) and then we need to save both to try again later (despite writing this code 😅).

I see that we have a doc comment above the definitions of `attempt`, `batch` etc that explains this -- maybe move this definition up there?